src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -441,6 +453,201 @@ try
 }
 CATCH_RETURN()
 
+// Routine Description:
+// - Returns whether the user set or updated any of the font features to be applied
+bool DxFontRenderData::DidUserSetFeatures() const noexcept
+{
+    return _didUserSetFeatures;
+}
+
+// Routine Description:
+// - Updates our internal map of font features with the given features
+// - NOTE TO CALLER: Make sure to call UpdateFont after calling this for the feature changes
+//   to take place
+// Arguments:
+// - features - the features to update our map with
+void DxFontRenderData::SetFeatures(const std::unordered_map<std::wstring_view, uint32_t>& features)

```

Comment:
This is only called when there _are_ features, right? So there's no chance that _featureVector will be empty but we try to get stuff out of it?

Reply:
This will get called even if features is empty, because we have no way of knowing if the user deleted features. _featureVector will always have at least the default features